File: test/test_decision_tree.rb
$: << File.join(File.dirname(__FILE__), "..", "lib")

require 'test/unit'
require 'brains'

class DecisionTreeTest < Test::Unit::TestCase

  def test_it_classifies
    d = Brains::DataFile.new("data/mushroom.csv")
    test, train = d.split(4, true)
    t = Brains::DecisionTree.new(train)
    correct = 0
    test.each do |klass,features|
      c = t.classify(features)
      correct += 1 if klass == c
    end
    assert correct.to_f / test.size >= 0.9, "bad classify"
  end

end

File: tests/test_decision_tree.py
from __future__ import division

import unittest
import brains

class DecisionTreeTestCase(unittest.TestCase):

    def test_classify(s):
        f = brains.DataFile("data/mushroom.csv")
        test, train = f.split(4)
        d = brains.DecisionTree(f.features, train)
        correct = 0
        for t in test:
            c = d.classify(t[1])
            if c == t[0]:
                correct += 1
        assert correct / len(test) > 0.9
